src/modules/visits/queries/find-actual-visit-by-user-and-space.graphql

```diff
@@ -3,10 +3,7 @@ query findActualVisitByUserAndSpace($userProfileId: uuid!, $spaceSlug: String!,
 		where: {
 			user_profile_id: { _eq: $userProfileId }
 			visitor_space: { slug: { _eq: $spaceSlug } }
-			_or: [
-				{ start_date: { _lte: $now }, end_date: { _gte: $now } }
-				{ status: { _eq: "PENDING" } }
-			]
+			_or: [{ end_date: { _gte: $now } }, { status: { _eq: "PENDING" } }]
```

Comment:
This might make it easier to check in code if you only fetch pending or approved visit requests.
Then you don't have to check for canceled or denied statuses in the code.

You might also want to rename the query at that point to: findPendingOrApprovedVisitRequestsForUser
```suggestion
			_or: [{ end_date: { _gte: $now }, status: { _eq: "APPROVED" } }, { status: { _eq: "PENDING" } }]
```

Reply:
👍